src/ray/util/event.cc

```diff
@@ -417,22 +430,35 @@ void RayExportEvent::SendEvent() {
 
 static absl::once_flag init_once_;
 
-void RayEventInit(rpc::Event_SourceType source_type,
+void RayEventInit(const std::vector<SourceTypeVariant> source_types,
                   const absl::flat_hash_map<std::string, std::string> &custom_fields,
                   const std::string &log_dir,
                   const std::string &event_level,
                   bool emit_event_to_log_file) {
   absl::call_once(
       init_once_,
-      [&source_type, &custom_fields, &log_dir, &event_level, emit_event_to_log_file]() {
-        RayEventContext::Instance().SetEventContext(source_type, custom_fields);
-        auto event_dir = std::filesystem::path(log_dir) / std::filesystem::path("events");
-        ray::EventManager::Instance().AddReporter(
-            std::make_shared<ray::LogEventReporter>(source_type, event_dir.string()));
+      [&source_types, &custom_fields, &log_dir, &event_level, emit_event_to_log_file]() {
+        for (const auto &source_type : source_types) {
+          std::string source_type_name = "";
+          if (auto event_source_type_ptr =
+                  std::get_if<rpc::Event_SourceType>(&source_type)) {
+            // Set custom fields for non export events
+            RayEventContext::Instance().SetEventContext(
+                std::get<rpc::Event_SourceType>(source_type), custom_fields);
+            source_type_name = Event_SourceType_Name(*event_source_type_ptr);
+          } else if (auto export_event_source_type_ptr =
+                         std::get_if<rpc::ExportEvent_SourceType>(&source_type)) {
+            // For export events
+            source_type_name = ExportEvent_SourceType_Name(*export_event_source_type_ptr);
+          }
+          auto event_dir =
+              std::filesystem::path(log_dir) / std::filesystem::path("events");
+          ray::EventManager::Instance().AddReporter(
```

Comment:
I'd like to suggest to have a separate map of reporters for export and non-export case instead of adding if statement to `ReportExportEvent ` and `Report` API

Reply:
Sounds good, updated